homeassistant/components/dlink/config_flow.py

```diff
@@ -51,7 +71,9 @@ async def async_step_user(
             step_id="user",
             data_schema=vol.Schema(
                 {
-                    vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+                    vol.Required(
+                        CONF_HOST, default=user_input.get(CONF_HOST, self.ip_address)
+                    ): str,
```

Comment:
We shouldn't give them the option to enter the host here since we already set the unique id above and if they enter a different one we will now have the wrong unique id